src/clr-addons/datagrid/enum-filter/enum-filter.component.ts

```diff
@@ -90,12 +98,24 @@ export class ClrEnumFilterComponent<T extends { [key: string]: any }>
   }
 
   private emitFilterChanged() {
-    this.clrFilterValuesChange.emit(this.filteredValues);
+    this.clrFilterValuesChange.emit(this.getDisplayValues(this.filteredValues));
     this.changes.emit(true);
   }
 
   ngOnDestroy() {
     this.destroyed$.next();
     this.destroyed$.complete();
   }
+
+  protected containsFilterValue(entries: FilterValue[], filtered: FilterValue): boolean {
+    return !!entries.find(v => v.value === filtered.value);
```

Comment:
you could use "entries.some(..)" which directly returns a boolean

Reply:
done